src/main/java/httpserver/httpmessages/HTTPResponseParser.java

```diff
@@ -0,0 +1,49 @@
+package httpserver.httpmessages;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class HTTPResponseParser {
+
+    private static final String PROTOCOL_VERSION = "HTTP/1.1";
+    private static final String CONTENT_TYPE = "Content-Type : ";
+    private static final String SPACE = " ";
+    private final ByteArrayOutputStream byteArrayOutputStream;
+
+    public HTTPResponseParser(ByteArrayOutputStream byteArrayOutputStream) {
+        this.byteArrayOutputStream = byteArrayOutputStream;
+    }
+
+    public byte[] parse(HTTPResponse httpResponse) {
+        addHeader(httpResponse, byteArrayOutputStream);
+        if (httpResponse.hasBody()) {
+            addBody(httpResponse, byteArrayOutputStream);
+        }
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    private void addBody(HTTPResponse httpResponse, ByteArrayOutputStream byteArrayOutputStream) {
+        try {
+            byteArrayOutputStream.write(getContentType(httpResponse));
+            byteArrayOutputStream.write(httpResponse.getBody());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void addHeader(HTTPResponse httpResponse, ByteArrayOutputStream byteArrayOutputStream) {
+        try {
+            byteArrayOutputStream.write(getHeader(httpResponse));
+        } catch (IOException e) {
+            throw new ByteWriterException("Cannot write header : ", e);
```

Comment:
Nooooooooooooo please please please don't.  Unless you do it eeeeeeeverywhere.
The keyword final in Java is much less strict than one would think. It only means we can't change the reference, but we can change the content behind the reference.
I accept that putting final in the parameters catches if you from re-assigning to them, but short methods make it obvious when it happens.
I favour  "less visual noise" over "stops me from reassigning"